x-pack/legacy/plugins/integrations_manager/index.ts

```diff
@@ -74,7 +74,14 @@ const pluginOptions: LegacyPluginOptions = {
   // yes, any. See https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/infra/server/lib/adapters/configuration/kibana_configuration_adapter.ts#L49-L58
   // for a way around it, but this is Legacy Platform and I'm not sure these hoops are worth jumping through.
   init(server: any) {
-    server.plugins.xpack_main.registerFeature(feature);
+    const { newPlatform } = server;
+    const npSetup = newPlatform.setup;
+    const featuresPlugin = npSetup.plugins.features;
+
+    if (!featuresPlugin) {
+      throw new Error('New Platform XPack Features plugin is not available.');
```

Comment:
throwing from init will crash the entire kibana server. There is no NP equivalent, but in legacy I believe the best thing to do when your plugin can't start up is to signal that by setting it's status to red with `this.status.red(err)`.